R/rj.b.R

```diff
@@ -165,13 +172,39 @@ RjClass <- R6::R6Class(
 
             } else if (os == 'Windows') {
 
-                entries <- try(readRegistry('Software\\R-core\\R', 'HLM'))
-                if (inherits(entries, 'try-error'))
+                regHLM <- file.path("SOFTWARE", "R-core", "R64", fsep = "\\")
+                entries <- try(readRegistry(regHLM, 
+                                            hive = "HLM",
+                                            maxdepth = 2, 
+                                            view = "64-bit"))
+
+                if ( inherits(entries, 'try-error') )
                     stop('Could not find system R (no appropriate registry entries)')
-                home <- entries$InstallPath
-                path <- file.path(home, 'bin', 'x64', 'R.exe')
-                if (file.exists(path))
-                    return(path)
+
+                if ( is.null(entries$`Current Version`) || is.null(entries$InstallPath) ) {
+
+                    ssDx <- function(x, n) {sapply(x, function(xx) substr(xx, (nchar(xx)-n+1), nchar(xx)) )}
+
+                    path <- file.path(entries[[1]]$InstallPath)
+                    rcv <- ssDx(path, 5)
+
+                    path <- file.path(path, 
+                                      'bin', 
+                                      'x64', 
+                                      'R.exe')
+                } else {
+                    rcv <- entries$`Current Version`
+                    path <- file.path(entries$InstallPath, 'bin', 'x64', 'R.exe')
+                }
+
+                path <- gsub('/', '\\', path, fixed=TRUE)
+
+                if ( file.exists(path) ){
+                    
+                    results <- list(path=path, rcv=rcv)
+                    return(results)
```

Comment:
so i gather you've changed the signature of .findR() so that it returns a list(path=..., rcv=...) which is fine, but it now has different behaviour depending on whether it's under windows or linux ... let's try and keep the behaviour somewhat consistent ... so maybe change it so it returns `list(path=...)` under linux and macos too (you can leave out the `rcv` for macos and linux if we never use it)

if functions have different return types based on context, it can end up being a trap for the next person coming along.

Reply:
>which is fine, but it now has different behaviour depending on whether it's under windows or linux ... 
 
 Yes, you are right!